gobblin-utility/src/main/java/gobblin/util/logs/LogCopier.java

```diff
@@ -407,23 +443,49 @@ public LogCopier build() {
   private class LogCopyTask implements Runnable {
 
     private final Path srcLogFile;
-    private final Path destLogFile;
+    private final Path destLogDir;
+    private final String destLogFileBaseName;
+    private final String destLogFileExtension;
+    private final Stopwatch watch = Stopwatch.createStarted();
+
+    private Path currentDestLogFile;
 
     // The task maintains the current source log file position itself
     private long currentPos = 0;
 
     LogCopyTask(Path srcLogFile, Path destLogFile) {
       this.srcLogFile = srcLogFile;
-      this.destLogFile = destLogFile;
+      this.destLogDir = destLogFile.getParent();
+      this.destLogFileBaseName = FilenameUtils.removeExtension(destLogFile.getName());
+      this.destLogFileExtension = FilenameUtils.getExtension(destLogFile.getName());
+      this.currentDestLogFile = getCurrentDestLogFile();
+      this.watch.start();
+    }
+
+    private Path getCurrentDestLogFile() {
+      return new Path(this.destLogDir,
+          this.destLogFileBaseName + "." + System.currentTimeMillis() + "." + this.destLogFileExtension);
```

Comment:
I think a better approach is to always name the current log file as normal like `GobblinApplicationMaster.stdout` and rename it to something like `GobblinApplicationMaster.stdout.<timestamp>` when necessary and re-create a new empty file named `GobblinApplicationMaster.stdout`.